DigitalCrafts/Week4/JavascriptExercises/JS_higherFn_jasmine/scripts/main.js

```diff
@@ -0,0 +1,69 @@
+
+function isPositive(n){
+    return n > 0;
+}
+function isEven(n){
+    return n % 2 === 0;
+}
+function squareIt(n){
+    return n * n;
+}
+function coolCity(obj){
+    return obj.temperature < 70;
+}
+function returnCityName(obj){
+    return obj.name;
+}
+function printGreeting(person){
+    return ("hello " + person);
+}
+function productMaker(a, b){
+    return a * b;
+}
+function returnTotal(array){
+    return array.map(returnPrice).reduce((a,b)=> a + b);
+
+}
+
+function returnPrice(obj){
+    return obj.price
+}
+function addSep(array, sep){
+    function derp(currentValue, index, array){
```

Comment:
I did an additional exercise to refactor this and left the original as an example of how to make things overly complicated